### DISCUSSION TITLE
Sort by relation, where the relation is filtered before.

### DISCUSSION THREAD

**Discussion by danielyogel** [prisma/prisma#17345] [Q&A]
Hi all,

How may I write a query, which sorts by relation, but before It filters the related table ?

Something like:

`const getActiveUsers = await prisma.user.findMany({
  take: 10,
  orderBy: {
    posts: {
      _count: 'desc',
      where: {postKind: 'BLOG_POST'}
    },
  },
})`

Thank you!

**Comment by nurul3101** [comment]
Hello @danielyogel 👋 
Can you share your schema file (Particularly user and post models) so I can better understand what you are trying to achieve?

**Comment by danielyogel** [comment]
Helo @nurul3101,
Yes, something like this.
Thank you!

```
model User {
  id          String    @id @default(cuid())
  content     String
  followedBy  Pool[]    @relation("following")
  following   Pool[]    @relation("follower")
}

model Pool {
  kind        PoolKind
  follower    User    @relation("follower", fields: [followerId], references: [id], onDelete: Cascade)
  followerId  String
  following   User    @relation("following", fields: [followingId], references: [id], onDelete: Cascade)
  followingId String
  @@id([followerId, followingId, kind])
}

enum PoolKind {
  RELATION_ONE
  RELATION_TWO
}
```

`const getActiveUsers = await prisma.user.findMany({ take: 10, orderBy: { followedBy: { _count: 'desc', where: {kind: 'RELATION_TWO'} }, }, })`

**Reply by nurul3101** [reply]
Apologies for the delay in getting back to you, this conversation slipped through the cracks.
Can you try this query?
```typescript
  const getActiveUsers = await prisma.user.findMany({
    where: {
      followedBy: {
        some: {
          kind: 'RELATION_ONE',
        },
      },
    },
    take: 10,
    orderBy: {
      followedBy: { _count: 'desc' },
    },
  });
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
